src/components/UASTViewer.js

```diff
@@ -1,18 +1,35 @@
 import React, { Component } from 'react';
 import styled from 'styled-components';
+import Node from './uast/Node';
+import {eventHandler} from './uast/EventHandler';
+import {font, background} from '../styling/variables';
 
 const Container = styled.div`
+  position: absolute;
+  height: 100%;
+  width: 100%;
+  overflow: auto;
+  font-family: ${font.family.code};
+  font-size: ${font.size.large};
+  background: ${background.main};
+  line-height: ${font.lineHeight.large};
+  padding: 4px;
 `
 
 export default class UASTViewer extends Component {
+  constructor(props) {
+    super(props);
+    eventHandler.onNodeSelected = this.props.onNodeSelected;
```

Comment:
Whose default export is the instance. That is, a singleton.
But that's not how things are done the react way. Parents pass callbacks down as props, children call them.